app/src/androidTest/java/com/github/polybooks/FilteringBooksTest.kt

```diff
@@ -0,0 +1,129 @@
+package com.github.polybooks
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.scrollTo
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasExtraWithKey
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.Matchers.allOf
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class FilteringBooksTest {
+    @get:Rule
+    val activityRule : ActivityScenarioRule<FilteringBooksActivity>
+            = ActivityScenarioRule(FilteringBooksActivity::class.java)
+
+    @Test
+    fun intentIsFiredWhenClickingOnResults() {
+        Intents.init()
```

Comment:
You could do the following to avoid code duplication
```
@Before
    fun before() {
        Intents.init()
    }

    @After
    fun after() {
        Intents.release()
    }
```